pkg/util/iptables/save_restore.go

```diff
@@ -72,6 +73,19 @@ func GetChainLines(table Table, save []byte) map[Chain][]byte {
 	return chainsMap
 }
 
+// GetExistingChains get iptables-save output so we can check for existing chains and rules.
+// This will be a map of chain name to chain with rules as stored in iptables-save/iptables-restore
+// Result may SHARE memory with contents of buffer.
+func GetExistingChains(iptables Interface, buffer *bytes.Buffer, table Table) map[Chain][]byte {
```

Comment:
I'm not super excited about this mish-mash of functions that are accumulating in this package.  Some read from save-state and some read directly.  We should probably define a type for saved data and move things like GetChainLines to methods there.  I guess that is a different conversation.

As an API, how about restructuring this like:

```
// If saveBuffer is provided, the saved data will be stored there, replacing any existing datat.  If saveBuffer is nil, a new buffer will be allocated.
func GetExistingChains(iptables Interface, table Table, saveBuffer *bytes.Buffer) map[Chain][]byte {
    if saveBuffer == nil {
        saveBuffer = bytes.NewBuffer()
    } else {
        saveBuffer.Reset()
    }
}
```

Reply:
This really starts to want a test.  Can you see how bad it would be?  It shouldn't be too hard - there are some examples in this dir using fake exec.